### PR TITLE
chore(common): cleanup build.inc.sh echo

### PR DESCRIPTION
**Pull request by mcdurdin**
Makes the output of log messages from builder scripts more consistent, and uses color more effectively to highlight specific parts of lines.

Note: updates documentation for builder_describe_internal_dependency from parent PR #8342.

Sample:
image

* Highlights top-most script with bright-white+bold identifier
* Dependencies have grey identifiers
* Blue for headings
* Purple for dependency/parameter info
* Yellow for warnings
* Red for errors

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by mcdurdin** [comment]
iOS build fail due to non-standard way of including builder.inc.sh:

image

Will fix the source -- we want to ensure that builder.inc.sh is always included correctly because otherwise `$THIS_SCRIPT` and other variables will not be defined correctly.

File: resources/builder.inc.sh [hunk 1]
```diff
@@ -143,17 +147,47 @@ function builder_heading() {
...
+  if [[ ! -z ${COLOR_RED+x} ]]; then
+    case $color in
+      white) color="$COLOR_WHITE" ;;
+      grey) color="$COLOR_GREY" ;;
+      green|success) color="$COLOR_GREEN" ;;
+      blue|heading) color="$COLOR_BLUE" ;;
+      yellow|warning) color="$COLOR_YELLOW" ;;
+      red|error) color="$COLOR_RED" ;;
+      teal) color="$COLOR_TEAL" ;;
+      setmark) mark="$HEADING_SETMARK" color="$COLOR_PURPLE" ;;
```
Comment (darcywong00) [inline resources/builder.inc.sh:169]: What about bright white and purple (separate from setmark)?

File: resources/builder.inc.sh [hunk 2]
```diff
@@ -808,11 +838,10 @@ _builder_parameter_error() {
   local program="$1"
   local type="$2"
   local param="$3"
-  echo "$COLOR_RED$program: invalid $type: $param$COLOR_RESET"
+  builder_echo red "$program: invalid $type: $param"
```
Comment (darcywong00) [inline resources/builder.inc.sh:841]: Generally, is `builder_die` used for `builder_echo red`?

File: resources/build/build-utils.md [hunk 3]
```diff
@@ -388,6 +408,35 @@ script. If not specified, dependencies will be built for all actions on all
...
+```bash
+builder_describe_internal_dependency \
+  mac:build mac-x86_64:build \
+  mac:build mac-arm64:build
+```
+
+**Note:** actions and targets must be fully specified, and this _must_ be called
+before either builder_describe_outputs or builder_parse in order for
+dependencies to be resolved.
+
```
Comment (darcywong00) [inline resources/build/build-utils.md:438]: Do the new colors also get added to https://github.com/keymanapp/keyman/blob/1bc31c8792b460233ab015c59567ef2d5e6bf966/resources/build/build-utils.md#formatting-variables ?

File: resources/builder.inc.sh (hunk 2 above)
Comment (mcdurdin) [reply resources/builder.inc.sh:841]: It's possible to have a script emit an error before failing. But generally, yes, `builder_die` is the most common way we would want to emit red 'error' text.

File: resources/builder.inc.sh (hunk 1 above)
Comment (mcdurdin) [reply resources/builder.inc.sh:169]: Can do

File: resources/build/build-utils.md (hunk 3 above)
Comment (mcdurdin) [reply resources/build/build-utils.md:438]: Yes, may be a good idea, perhaps under a new `builder_echo` topic

**Review by mcdurdin — COMMENTED** [review]
chore: address review comments

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.62-alpha](https://keyman.com/downloads/releases/alpha/17.0.62)

**Pull request merged**